blackduck/Utils.py

```diff
@@ -14,38 +14,35 @@
 logger = logging.getLogger(__name__)
 
 
-def iso8601_to_date(iso_string, with_zone=False):
-    """Utility function to convert iso_8601 formatted string to datetime object, optionally accounting for timezone
+def to_datetime(date):
+    """Utility function to convert common date formats to datetime object
 
     Args:
-        iso_string (string): the iso_8601 string to convert to datetime object
-        with_zone (bool, optional): whether to account for timezone offset. Defaults to False.
+        iso_string (string/datetime/date)
 
     Returns:
-        datetime.datetime: equivalent time, with or without timezone offsets
+        datetime.datetime
     """
-    date_timezone = iso_string.split('Z')
-    date = dateutil.parser.parse(date_timezone[0])
-    if with_zone and len(date_timezone > 1):
-        hours_minutes = date_timezone[1].split(':')
-        minutes = (60*int(hours_minutes[0]) + int(hours_minutes[1] if len(hours_minutes) > 1 else 0))
-        date = date + datetime.timedelta(minutes=minutes)
-    return date
-
-def iso8601_timespan(days_ago, from_date=datetime.utcnow(), delta=timedelta(weeks=1)):
+    if isinstance(date, str):
+        from dateutil.parser import parse
+        return parse(date)
+    if isinstance(date, datetime.datetime):
+        return date
+    if isinstance(date, datetime.date):
+        return datetime.datetime(
+            year=date.year,
+            month=date.month,
+            day=date.day
+        )
+    raise TypeError(f"object of type {type(date)} cannot be converted to datetime")
+        
+
+def timespan(days_ago, from_date=datetime.now(), delta=timedelta(weeks=1)):
     curr_date = from_date - timedelta(days=days_ago)
     while curr_date < from_date:
-        yield curr_date.isoformat('T', 'seconds')
+        yield curr_date
         curr_date += delta
 
-def min_iso8601():
```

Comment:
While I am all for deleting code, are these methods that you just recently added e.g. this year? Generally, deleting or changing method signatures are considered as breaking backwards-compatibility in an API.

Reply:
None of these were in HubInstance - the only use they had was in demo_client, though I'll update that to use the two new methods.

Reply:
Okay I'm fine with deleting them.